x/txfees/keeper/txfee_filters/arb_tx.go

```diff
@@ -160,3 +185,68 @@ func isArbTxLooseSwapMsg(swapMsg poolmanagertypes.SwapMsgRoute, swapInDenom stri
 	swapInDenom = swapMsg.TokenInDenom()
 	return swapInDenom, false
 }
+
+func isWasmHooksExecutePayload(memo string) (map[string]interface{}, bool) {
+	jsonObject := make(map[string]interface{})
+	err := json.Unmarshal([]byte(memo), &jsonObject)
+	if err != nil {
+		return nil, false
+	}
+	wasm, ok := jsonObject["wasm"].(map[string]interface{})
+	if !ok {
+		return nil, false
+	}
+	_, ok = wasm["contract"].(string)
+	if !ok {
+		return nil, false
+	}
+	msg, ok := wasm["msg"].(map[string]interface{})
+	if !ok {
+		return nil, false
+	}
+	return msg, true
+}
+
+func getLoalIBCDenom(packet channeltypes.Packet, denom string) string {
+	if transfertypes.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel(), denom) {
+		// sender chain is not the source, unescrow tokens
+
+		// remove prefix added by sender chain
+		voucherPrefix := transfertypes.GetDenomPrefix(packet.GetSourcePort(), packet.GetSourceChannel())
+		unprefixedDenom := denom[len(voucherPrefix):]
```

Comment:
is it 100% sure that `(len(voucherPrefix < len(denom)`?

Reply:
It should be. This is all what IBC does internally, I just copied that code.